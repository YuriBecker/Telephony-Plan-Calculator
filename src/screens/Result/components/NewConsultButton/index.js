import React from 'react';
import { Button } from '@ui-kitten/components';
import PropTypes from 'prop-types';
import { ViewPropTypes } from 'react-native';

const index = ({ onPress, children, size, style }) => {
  return (
    <Button onPress={onPress} size={size} style={style}>
      {children}
    </Button>
  );
};

export default index;

index.propTypes = {
  onPress: PropTypes.func.isRequired,
  size: PropTypes.string,
  children: PropTypes.node.isRequired,
  style: ViewPropTypes.style,
};

index.defaultProps = {
  size: 'large',
  style: {},
};
