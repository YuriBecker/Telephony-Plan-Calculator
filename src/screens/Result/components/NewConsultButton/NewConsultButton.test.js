import React from 'react';
import { shallow } from 'enzyme';

import NewConsultButton from './index';

describe('Testing NewConsultButton component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <NewConsultButton onPress={jest.fn()}>Some text</NewConsultButton>
    );
  });

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
});
