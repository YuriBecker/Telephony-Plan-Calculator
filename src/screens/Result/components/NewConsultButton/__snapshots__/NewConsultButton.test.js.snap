// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing NewConsultButton component should render correctly 1`] = `
<ButtonComponent
  onPress={[MockFunction]}
  size="large"
  style={Object {}}
>
  Some text
</ButtonComponent>
`;
